code/obj/machinery/waterslide.dm

```diff
@@ -0,0 +1,151 @@
+// legally distinct water slides! deploy a line of slippery tiles that apply touch chems! wowie!
+
+/obj/machinery/waterslide_pump
+	name = "water slide pump"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "waterslide-pump"
+	desc = "A pump and hose attached to a big roll of plastic."
+	density = 0 // Being able to step through this makes the 'sliding' part more intuitive
+	anchored = 1
+
+	var/active = FALSE
+
+	var/steps = 7
+	var/max_steps = 15
+	var/min_steps = 1
+	var/current_step = 1
+
+	var/list/deployed_path = new/list()
+
+	var/obj/item/fluidtank = null
+
+	New()
+		..()
+
+		fluidtank = new/obj/item/reagent_containers/food/drinks/fueltank/chlorine/mostly_water(src)
+
+	disposing()
+		for(var/obj/item/reagent_containers/waterslide/slide_tile in deployed_path)
+			slide_tile.dispose()
+
+		..()
+
+	process()
+		if(active)
+			if(current_step > deployed_path.len)
+				current_step = 1
+			if(!fluidtank || fluidtank.reagents.total_volume <= 0)
+				active = FALSE
+				return
+			var/obj/item/reagent_containers/waterslide/section = deployed_path[current_step]
+
+			if(section.reagents.total_volume <= 0)
+				fluidtank.reagents.trans_to(section,5)
+			if(src.fluidtank.reagents.total_volume <= 0)
+				active = FALSE
+				current_step = 1
+				return
+
+			current_step += 1
+
+			SPAWN_DBG(2 SECONDS) src.process()
```

Comment:
instead of this, just let the machinery process loop take care of this. It will then also take care of things like power consumption. 
 Look at other instances of machinery for implementation 
 I think the machinery loop is about 2 seconds anyway